src/envoy/http/authn/http_filter.h

```diff
@@ -15,18 +15,25 @@
 
 #pragma once
 
+#include <memory>
 #include "authentication/v1alpha1/policy.pb.h"
 #include "common/common/logger.h"
 #include "server/config/network/http_connection_manager.h"
+#include "src/envoy/http/authn/jwt_authn_store.h"
+#include "src/envoy/http/jwt_auth/jwt_authenticator.h"
 
 namespace Envoy {
 namespace Http {
 
 // The authentication filter.
 class AuthenticationFilter : public StreamDecoderFilter,
-                             public Logger::Loggable<Logger::Id::filter> {
+                             public JwtAuth::JwtAuthenticator::Callbacks,
+                             public Logger::Loggable<Logger::Id::http> {
  public:
-  AuthenticationFilter(const istio::authentication::v1alpha1::Policy& config);
+  AuthenticationFilter(const istio::authentication::v1alpha1::Policy& config,
+                       Upstream::ClusterManager& cm,
+                       IstioAuthn::JwtMultiFactoryStore& jwt_store);
```

Comment:
Should this be a map/vector of JwtAuthStore, instead of Factory?

Reply:
Because JwtAuthStoreFactory has included the thread local related implementations whereas JwtAuthStore does not have thread local related implementations, this should be a map/vector of JwtAuthStoreFactory, instead of a map/vector of JwtAuthStore.

Reply:
Look at http_filter from jwt_auth (https://github.com/istio/proxy/blob/master/src/envoy/http/jwt_auth/http_filter.h#L31), it takes JwtAuthStore object (ref). JwtAuthStoreFactory is created in http_filter_factory, but is used only to pass the store(). I don't see why do you need to the whole (array of) factory here.